taichi/codegen/spirv/spirv_codegen.cpp

```diff
@@ -1828,10 +1834,18 @@ class TaskCodegen : public IRVisitor {
 
     std::vector<std::tuple<spirv::SType, std::string, size_t>>
         struct_components_;
+    // Now we only have one ret
     for (auto &ret : ctx_attribs_->rets()) {
-      struct_components_.emplace_back(ir_->get_primitive_type(ret.dt),
-                                      "ret" + std::to_string(ret.index),
-                                      ret.offset_in_mem);
+      if (auto tensor_type = ret.dt->cast<TensorType>()) {
+        struct_components_.emplace_back(
+            ir_->get_array_type(ir_->i64_type(),
```

Comment:
Why always i64?

Reply:
To match taichi's result buffer, now it only has 30 slots of uint64 type. For convenience, I try to cast the return value to i64.

Reply:
Maybe should consider using / modifying the host device contex blitter

Reply:
fix